utils/spdx/src/test/kotlin/SpdxExpressionTest.kt

```diff
@@ -175,6 +175,46 @@ class SpdxExpressionTest : WordSpec() {
             }
         }
 
+        "An SpdxExpression with LicenseRefs" should {
+            val licenseRefExpression = "LicenseRef-ort-license"
+            val licenseRefExpressionWithException = "$licenseRefExpression WITH LicenseRef-ort-exception"
+            val licenseRefExpressionWithException2 = "$licenseRefExpression WITH LicenseRef-ort-exception-2.0"
+            val licenseRefExpressionWithLicenseRefExpression = "$licenseRefExpression WITH $licenseRefExpression"
+
+            "be valid in lenient mode" {
+                assertSoftly {
+                    licenseRefExpression.toSpdx(Strictness.ALLOW_ANY)
+                    licenseRefExpressionWithException.toSpdx(Strictness.ALLOW_ANY)
+                    licenseRefExpressionWithException2.toSpdx(Strictness.ALLOW_ANY)
+                    licenseRefExpressionWithLicenseRefExpression.toSpdx(Strictness.ALLOW_ANY)
+                }
+            }
+
+            "be valid in deprecated mode" {
+                assertSoftly {
+                    licenseRefExpression.toSpdx(Strictness.ALLOW_DEPRECATED)
+                    licenseRefExpressionWithException.toSpdx(Strictness.ALLOW_DEPRECATED)
+                    licenseRefExpressionWithException2.toSpdx(Strictness.ALLOW_DEPRECATED)
```

Comment:
I understood the code so that with `ALLOW_DEPRECATED` and `ALLOW_CURRENT` only SPDX exception identifiers are allowed. This would also be according with the spec (see [D.4.4](https://spdx.github.io/spdx-spec/SPDX-license-expressions/#d44-exception-with-operator)):

> If the applicable exception is not found on the SPDX License Exception List, then use a single <license-ref> to represent the entire license terms (including the exception).

Reply:
I don't really understand why custom exceptions are not allowed there, but it's also in the draft for SPDX v3: https://spdx.github.io/spdx-spec/v3-draft/SPDX-license-expressions/#d44-exception-with-operator

Reply:
There is an open issue to change that for v3: https://github.com/spdx/spdx-spec/issues/153

Reply:
I've rewritten the code to introduce a new `ALLOW_LICENSEREF_EXCEPTIONS` strictness level instead.